app_feup/lib/view/Pages/BusStopSelectionPage.dart

```diff
@@ -0,0 +1,263 @@
+
+  List<Widget> getStopsTextList() {
+    List<Widget> stops = new List();
+    for (BusStop stop in configuredStops) {
+      stops.add(Text(stop.stopCode));
+    }
+    return stops;
+  }
+
+  Future deleteStop(BuildContext context, BusStop stop) async {
+    await db.removeBusStop(stop);
+    StoreProvider.of<AppState>(context).dispatch(setUserBusStops(new Completer()));
+  }
+
+  @override
+  Widget getBody(BuildContext context) {
+    return StoreConnector<AppState, List<dynamic>>(
+      converter: (store) => store.state.content['busstops'],
+      builder: (context, busStops) {
+        this.configuredStops = busStops;
+        return ListView(
+            children: <Widget>[
+              Container(
+                  padding: EdgeInsets.only(bottom: 12.0, right: 22.0),
+                  child: PageTitle(name: 'Paragens Configuradas')
+              ),
+              Column(
+                  children: List.generate(getStopsTextList().length, (i) {
+                    return Container(
+                        padding: EdgeInsets.only(top: 8.0, bottom: 8.0, left: 40.0, right: 40.0),
+                        child: RowContainer(
+                            child: Container(
+                                padding: EdgeInsets.only(left: 60.0, right: 60.0),
+                                child: Row(
+                                    mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                                    children: <Widget>[
+                                      getStopsTextList()[i],
+                                      IconButton(
+                                        icon: Icon(Icons.cancel),
+                                        color: darkGreyColor,
+                                        onPressed: () {
+                                          deleteStop(context, configuredStops[i]);
+                                        },
+                                      )
+                                    ]
+                                )
+                            )
+                        )
+                    );
+                  })
+              ),
+              Align(
+                  alignment: Alignment.center,
+                  child: RaisedButton(
+                      child: Text("Adicionar"),
+                      onPressed: () {
+                        showSearch(context: context, delegate: BusStopSearch());
+                      }
+                  )
+              )
+
+            ]
+        );
+      },
+    );
+  }
+}
+
+class BusStopSearch extends SearchDelegate<String> {
+  List<String> suggestionsList = new List();
+  AppBusStopDatabase db;
+  BusStop stopToAdd;
+
+  BusStopSearch() {
+    this.getDatabase();
+  }
+
+  Future<void> getDatabase() async {
+    db = await AppBusStopDatabase();
+  }
+
+  @override
+  List<Widget> buildActions(BuildContext context) {
+    return [IconButton(icon: Icon(Icons.clear), onPressed: () {
+      query = "";
+    })];
+  }
```

Comment:
There doesn't seem to be any reason for this to return a list instead of just a Widget

Reply:
This is an override of the "buildActions" function of the SearchDelegate, which requires this function to return a list of actions.